### PR TITLE
Optionally specify partitions in configuration

### PR DESCRIPTION
**Pull request by louis-pie**
## Problem

Currently tap-kafka only supports consuming messages from all partitions in a topic. Processing of the messages is handled by single thread and also produces a single thread of singer messages. These single threads are bottlenecks when consuming from a kafka topic that uses multiple partitions for read and write scaling.

## Proposed changes

Create the ability to specify which partitions should be consumed by a tap. This will allow running multiple taps in parallel that consume from the same topic, improving message processing and singer message production.

If no partitions are specified, tap-kafka will still consume all partitions in the topic.

## Problem

Currently tap-kafka only supports consuming messages from all partitions in a topic. Processing of the messages is handled by single thread and also produces a single thread of singer messages. These single threads are bottlenecks when consuming from a kafka topic that uses multiple partitions for read and write scaling.

## Proposed changes

Create the ability to specify which partitions should be consumed by a tap. This will allow running multiple taps in parallel that consume from the same topic, improving message processing and singer message production.

If no partitions are specified, tap-kafka will still consume all partitions in the topic.

## Types of changes

What types of changes does your code introduce to PipelineWise?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] Description above provides context of the change
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Unit tests for changes (not needed for documentation changes)
- [x] CI checks pass with my changes
- [x] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [x] Commit message/PR title starts with `[AP-NNNN]` (if applicable. AP-NNNN = JIRA ID)
- [x] Branch name starts with `AP-NNN` (if applicable. AP-NNN = JIRA ID)
- [x] Relevant documentation is updated including usage instructions

## Test coverage

```
---------- coverage: platform linux, python 3.8.10-final-0 -----------
Name                                             Stmts   Miss  Cover
--------------------------------------------------------------------
tap_kafka/__init__.py                               79     18    77%
tap_kafka/common.py                                 18      0   100%
tap_kafka/errors.py                                 16      0   100%
tap_kafka/serialization/__init__.py                  0      0   100%
tap_kafka/serialization/json_with_no_schema.py      20      3    85%
tap_kafka/serialization/protobuf.py                 43      2    95%
tap_kafka/sync.py                                  248     12    95%
--------------------------------------------------------------------
TOTAL                                              424     35    92%

Required test coverage of 80% reached. Total coverage: 91.75%

```

### CODE REVIEW

File: tests/unit/test_tap_kafka.py [hunk 1]
```diff
@@ -1016,46 +946,46 @@ def test_iso_timestamp_to_epoch(self):
...
-        # TODO: This is just a fake unit test to reach the coverage limit, please complete it!
-        consumer = KafkaConsumerMock([])
-        sync.seek_partitions(consumer, [])
+    def test_epoch_to_iso_timestamp(self):
+        """Validate converting epoch milliseconds to ISO timestamp"""
+        self.assertEqual(sync.epoch_to_iso_timestamp(1635807671000),'2021-11-01T23:01:11.000')
+        self.assertEqual(sync.epoch_to_iso_timestamp(1635807671123),'2021-11-01T23:01:11.123')
+
+        # Invalid epoch should raise exception
+        with self.assertRaises(InvalidTimestampException):
```
Comment (amofakhar) [inline tests/unit/test_tap_kafka.py:955]: Based on the testing invalid values, I can't understand what we expect from the method if input value be `None`!
This test can be passed yet if the method be updated to return a value if input value be `None`
for example if I add this line at the beginning of the method:
```python
    if epoch is None:
        return datetime.datetime.utcfromtimestamp(1000000000)
```
_A good test should fail if the under test method is manipulated (by mistake, or ...etc)_

**Review by amofakhar — COMMENTED** [review]
Please check the codes by Flake8 and fix the issues, there are a lot of PEP8 problems.

File: tests/integration/test_protobuf.py [hunk 2]
```diff
@@ -81,7 +82,8 @@ def test_tap_kafka_protobuf(self):
         singer_messages = []
         singer.write_message = lambda m: singer_messages.append(m.asdict())
 
-        sync.do_sync(tap_kafka_config, catalog, state={'bookmarks': {topic: {}}})
+        sync.do_sync(tap_kafka_config, catalog, state={})
+        self.assertEqual(len(singer_messages), 6)
```
[the remark this answers is not in the document]
Comment (louis-pie) [reply tests/integration/test_protobuf.py:86]: Returning the incorrect amount of messages (off-by-one bug) is a common issue when dealing with consuming from kafka

Having this test fail is a much quicker indication of the issue than looking at the diff of the 6 messages. This technique is used extensively in tap-kafka.

File: tests/unit/test_tap_kafka.py (hunk 1 above)
Comment (louis-pie) [reply tests/unit/test_tap_kafka.py:954]: Added another test to fail if None is passed as a supposed epoch timestamp

**Review by amofakhar — APPROVED** [review]
